ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -436,6 +436,122 @@ object Trees {
     val tpe = AnyType
   }
 
+  /** Calls a method inherited from the parent class of `cls` on `receiver`.
+   *
+   *  `cls` must be a Scala.js-defined JS class.
+   *
+   *  Given the Scala.js-defined JS class
+   *
+   *  {{{
+   *  @ScalaJSDefined
+   *  class Foo extends Bar
+   *  }}}
+   *
+   *  The node
+   *
+   *  {{{
+   *  JSBrackerSuperCall(ClassType(Foo), receiver, method, args)
+   *  }}}
+   *
+   *  which is printed as
+   *
+   *  {{{
+   *  receiver.Foo::super[method](...args)
+   *  }}}
+   *
+   *  has the following semantics:
+   *
+   *  {{{
+   *  Bar.prototype[method].call(receiver, ...args)
+   *  }}}
+   *
+   *  If this happens to be located in an instance method of `Foo`, *and*
+   *  `receiver` happens to be `This()`, this is equivalent to the ES6
+   *  statement
+   *
+   *  {{{
+   *  super[method](...args)
+   *  }}}
+   */
+  case class JSBracketSuperCall(cls: ClassType, receiver: Tree, method: Tree,
+      args: List[Tree])(implicit val pos: Position) extends Tree {
+    val tpe = AnyType
+  }
```

Comment:
The idea is never to emit different IR, when talking to another class' public or protected API, depending on whether that class is native or not. Basically, being native or not is an implementation detail of the class wrt other IR classes manipulating it.